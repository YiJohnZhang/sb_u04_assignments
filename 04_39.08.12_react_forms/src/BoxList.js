import React, {useState} from 'react';
// import {v4 as uuidv4} from 'uuid';	
import Box from './Box';
import NewBoxForm from './NewBoxForm';

function BoxList(){

	const DEFAULT_BOXES = [{id: uuidv4(), width:10, height: 10, backgroundColor:'black'}];

	[boxes, setBoxes] = useState(DEFAULT_BOXES);

	function appendBox(boxParameters){


	}

	function removeBox(boxId){



	}

	return(
	<div>
		<NewBoxForm createNewBox={appendBox}/>
		{
			boxes.map((box) => (
				<Box
					key={box.id}
					width={box.width}
					height={box.height}
					backgroundColor={box.backgroundColor}
					/>
		))}

	</div>
	);

}