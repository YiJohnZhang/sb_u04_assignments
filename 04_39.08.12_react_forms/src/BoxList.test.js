import React from 'react';
import {render} from '@testing-library/react';
import BoxList from './BoxList';

test('smoke test', () => {

	render(<BoxList />);

});

test('snapshot', () => {
	
	const {asFragment} = render(<BoxList />);
	expect(asFragment()).toMatchSnapshot();

});

test('fireEvent (form): fields update', () => {
	const {asFragment} = render(<BoxList />);
	// probably keypress? or set the state?
	// test if the value has changed for the input

});

test('fireEvent (form): form submits', () => {

	// If not props, I am guess mocking or something to set the form initial state?

});

test('fireEvent (form): appends box', () => {

	// I am guess mocking or something to set the form initial state?

});

test('fireEvent (box): delete box', () => {

	// I am guess mocking or something to set the form initial state?

});