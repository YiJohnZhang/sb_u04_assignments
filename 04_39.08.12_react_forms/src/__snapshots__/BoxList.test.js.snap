// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        data-test-id="Component-input_width"
        name="width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
        Height: 
      </label>
      <input
        name="height"
        type="number"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Background Color: 
      </label>
      <input
        name="backgroundColor"
        type="text"
        value=""
      />
      <button
        data-test-id="submitButtonName"
      >
        Submit
      </button>
    </form>
    <div
      style="width: 100px; height: 100px; background-color: black;"
    >
      <button>
        x
      </button>
    </div>
  </div>
</DocumentFragment>
`;
